### PR TITLE
feat: remove `*.lido.fi` hardcode domains

### PR DESCRIPTION
**Pull request by solidovic**
### Description

Removed all hardcode domains of `*.lido.fi`.
Use env vars instead of.

### Testing notes

Check:
- lido logo (in header)
- meta preview img in ipfs version
- links in banner 'Whitelist mode is on'
- rpc block in ipfs version
- FAQ 'What is Lido staking APR for Ethereum?'
- Link in text 'Represents how much ether you can stake at this moment. You cannot stake over the global staking limit. The....'
- Goerli sunset banner
- Link add NFT GUIDE
- PENALTIES INFO LINK
- FAQ How can I get steth
- FAQ How can I unstake steth
- FAQ How can I use steth
- FAQ Lido Eth apr
- FAQ How can I get wsteth
- FAQ How can I use wsteth
- L2 banner
- ECOSYSTEM LINK in modal_pool_banners
- footet
- reef knot modal
  - termsLink
  - privacyNoticeLink

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# for supporting of domain agnostic
+SELF_ORIGIN=https://stake.lido.fi
+ROOT_ORIGIN=https://lido.fi
+DOCS_ORIGIN=https://docs.lido.fi
+HELP_ORIGIN=https://help.lido.fi
+RESEARCH_ORIGIN=https://research.lido.fi
```
Comment (Jeday) [inline .env.example:6]: Set those envs with right consts in ipfs jobs yamls

File: pages/_document.tsx [hunk 2]
```diff
@@ -68,7 +62,8 @@ export default class MyDocument extends Document {
   }
 
   get metaPreviewImgUrl(): string {
-    return `${host}/lido-preview.png`;
+    // TODO: needs fix for IPFS
+    return `${config.selfOrigin}/lido-preview.png`;
```
Comment (Jeday) [inline pages/_document.tsx:66]: We cannot know IPFS domain beforehand. It's okay to point to stake.lido.fi here

**Review by Jeday — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (solidovic) [reply .env.example:6]: done

File: pages/_document.tsx (hunk 2 above)
Comment (solidovic) [reply pages/_document.tsx]: done

**Review by manneredboor — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Review by DiRaiks — DISMISSED** [review]

**Review by Jeday — DISMISSED** [review]

**Review by alx-khramov — DISMISSED** [review]

File: features/stake/stake.tsx [hunk 3]
```diff
@@ -15,11 +15,11 @@ export const Stake = () => {
       <GoerliSunsetBanner />
       <NoSSRWrapper>
         <StakeForm key={key} />
+        <LidoStats />
```
Comment (Jeday) [inline features/stake/stake.tsx:18]: Why move into nossr wrapper? This content is static.

File: shared/components/layout/layout.tsx [hunk 4]
```diff
@@ -38,7 +39,9 @@ export const Layout: FC<PropsWithChildren<Props>> = (props) => {
         <LayoutSubTitleStyle>{subtitle}</LayoutSubTitleStyle>
         {children}
       </Main>
-      <Footer />
+      <NoSSRWrapper>
```
Comment (Jeday) [inline shared/components/layout/layout.tsx:42]: Why nossr footer?

**Review by Jeday — CHANGES_REQUESTED** [review]

File: shared/components/layout/layout.tsx (hunk 4 above)
Comment (solidovic) [reply shared/components/layout/layout.tsx]: revert

File: features/stake/stake.tsx (hunk 3 above)
Comment (solidovic) [reply features/stake/stake.tsx]: revert

**Review by Jeday — APPROVED** [review]

**Review by Jeday — APPROVED** [review]

**Review by jake4take — APPROVED** [review]

**Review by mvaleev — APPROVED** [review]

**Pull request merged**